x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/StringScriptFieldScript.java

```diff
@@ -49,12 +55,20 @@ public StringScriptFieldScript(Map<String, Object> params, SearchLookup searchLo
      */
     public final List<String> resultsForDoc(int docId) {
         results.clear();
+        chars = 0;
         setDocument(docId);
         execute();
         return results;
     }
 
     protected final void emitValue(String v) {
+        if (results.size() >= MAX_VALUES) {
+            throw new IllegalArgumentException("too many runtime values");
+        }
+        chars += v.length();
+        if (chars >= MAX_CHARS) {
+            throw new IllegalArgumentException("too many characters in runtime values [" + chars + "]");
```

Comment:
Runtime field [field] is emitting [1500] characters while the maximum number of characters allowed is [1000]"?